File: LeetCode/0001-two-sum/TwoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        for (i in nums.indices) {
            for (j in (i + 1) until nums.size) {
                val valI = nums[i]
                val valJ = nums[j]
                if (valI + valJ == target) return intArrayOf(i, j)
            }
        }
        return intArrayOf(0 , 1)
    }
}

File: LeetCode/0001-two-sum/two-sum.js
// Time complexity O(n), space complexity O(n)

var twoSum = function(nums, target) {

    const cache = {}
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i]
        if (complement in cache) {
            return [cache[complement], i]
        }
        cache[nums[i]] = i
    }
}
